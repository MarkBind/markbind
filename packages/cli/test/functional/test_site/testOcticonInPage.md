<frontmatter>
title: Test optimize loading of icon stylesheets
layout: simpleLayoutWithoutIcons.md
</frontmatter>


<quiz>
  <div slot="intro">
   There is an octicon icon in this quiz.
  </div>

  <question type="checkbox" header="There is an octicon icon in this question." hint="An octicon icon is supposed to appear here ---> :octicon-git-pull-request:">
  </question>
</quiz>

**There is one octicon icon in this page content.**

**Open up console. Only octicon stylesheet should be loaded!**
