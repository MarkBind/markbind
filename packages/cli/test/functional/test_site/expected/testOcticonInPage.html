<!DOCTYPE html>
<html>

<head>

  <meta name="default-head-top">

  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="MarkBind 5.4.0">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>octicon icon in page, only octicon stylesheet should be loaded</title>
  <link rel="stylesheet" href="/test_site/markbind/css/bootstrap.min.css">
  <link rel="stylesheet" href="/test_site/markbind/css/octicons.css">
  <link rel="stylesheet" href="/test_site/markbind/css/codeblock-dark.min.css">
  <link rel="stylesheet" href="/test_site/markbind/css/markbind.min.css">
  <link rel="stylesheet" href="/test_site/plugins/testMarkbindPlugin/testMarkbindPluginStylesheet.css">
  <link rel="stylesheet" href="/test_site/plugins/web3Form/web-3-form.css">
  <link rel="stylesheet" href="/test_site/plugins/markbind-plugin-anchors/markbind-plugin-anchors.css">
  <link rel="stylesheet" href="/test_site/plugins/markbind-plugin-tree/markbind-plugin-tree.css">

  <meta name="default-head-bottom">
  <link rel="stylesheet" href="/test_site/stylesheets/styles.css">
  <link rel="icon" href="/test_site/favicon.png">
</head>
<script>
  const baseUrl = '/test_site'
</script>

<body>
  <div id="app">

    <div>
      <header>
        <navbar type="dark" default-highlight-on="sibling-or-child">
          <template #brand><a href="/" title="Home" class="navbar-brand">MarkBind Test Site</a></template>
          <li><a class="nav-link" href="/test_site/bugs/index.html">Open Bugs</a></li>
        </navbar>
        <div class="bg-info display-4 text-center text-white">
          <br>
          Test Jumbotron<br>
          <br>
        </div>
      </header>
      <p><strong>Relative Link Test</strong> This is a relative Intra-Site link in a layout (see <a href="/test_site/index.html#heading-with-hidden-keyword">link</a>)</p>
    </div>
    <div id="flex-body">
      <overlay-source id="site-nav" tag-name="nav" to="site-nav">
        <div class="nav-component slim-scroll"></div>
      </overlay-source>
      <div id="content-wrapper">
        <breadcrumb></breadcrumb>

        <quiz>
          <template #intro>
            <div>
              There is an octicon icon in this quiz.
            </div>
          </template>
          <question type="checkbox"><template #hint>
              <p>An octicon icon is supposed to appear here ---&gt; <svg version="1.1" width="16" height="16" viewBox="0 0 16 16" class="octicon octicon-git-pull-request" aria-hidden="true">
                  <path fill-rule="evenodd" d="M7.177 3.073L9.573.677A.25.25 0 0110 .854v4.792a.25.25 0 01-.427.177L7.177 3.427a.25.25 0 010-.354zM3.75 2.5a.75.75 0 100 1.5.75.75 0 000-1.5zm-2.25.75a2.25 2.25 0 113 2.122v5.256a2.251 2.251 0 11-1.5 0V5.372A2.25 2.25 0 011.5 3.25zM11 2.5h-1V4h1a1 1 0 011 1v5.628a2.251 2.251 0 101.5 0V5A2.5 2.5 0 0011 2.5zm1 10.25a.75.75 0 111.5 0 .75.75 0 01-1.5 0zM3.75 12a.75.75 0 100 1.5.75.75 0 000-1.5z" />
                </svg></p>
            </template><template #header>
              <p>There is an octicon icon in this question.</p>
            </template></question>
        </quiz>
        <p><strong>There is one octicon icon in this page content.</strong></p>
        <p><strong>Open up console. Only octicon stylesheet should be loaded!</strong></p>
      </div>
      <overlay-source id="page-nav" tag-name="nav" to="page-nav">
        <div class="nav-component slim-scroll">
        </div>
      </overlay-source>
    </div>
  </div>
</body>
<script src="/test_site/markbind/js/bootstrap-utility.min.js"></script>
<script src="/test_site/markbind/js/polyfill.min.js"></script>
<script src="/test_site/markbind/js/vue.min.js"></script>
<script src="/test_site/markbind/js/markbind.min.js"></script>
<script src="testOcticonInPage.page-vue-render.js"></script>
<script>
  MarkBind.setupWithSearch()
</script>
<script src="/test_site/plugins/testMarkbindPlugin/testMarkbindPluginScript.js"></script>
<script>alert("Inline plugin script loaded!")</script>

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=TRACKING-ID"></script>
<script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());
    
      gtag('config', 'TRACKING-ID');
    </script>

<script>
        function submitForm(element) {
            event.preventDefault();
            const formData = new FormData(element);
            const formProps = Object.fromEntries(formData);
            const json = JSON.stringify(formProps);
            const submitButton = element.querySelector('button[type="submit"]');
            const submitButtonText = submitButton.innerText;
            submitButton.innerText = 'Please wait...'
            fetch('https://api.web3forms.com/submit', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                    'Accept': 'application/json'
                },
                body: json,
            })
            .then(async (response) => {
                if (response.status == 200) {
                    alert('Form submitted! Thank you for your response');
                } else {
                    alert('Error submitting form! Please try again later.');
                }
            })
            .catch(error => {
                alert('Error submitting form! Please try again later.');
            })
            .finally(() => {
                submitButton.innerText = submitButtonText;
            })
        }
    </script>
<script type="module">
    import mermaid from 'https://unpkg.com/mermaid@10/dist/mermaid.esm.min.mjs';
    document.addEventListener('DOMContentLoaded', () => {
      mermaid.initialize({});
      Vue.directive('mermaid', {
        inserted: function(el) {
          mermaid.run({
            nodes: [el]
          });
        }
      });
    });
  </script>

</html>